src/TeamSyncer.ts

```diff
@@ -392,17 +393,29 @@ export class TeamSyncer {
         const ghosts = await Promise.all(members.members.map(async(slackUserId) => this.main.ghostStore.get(slackUserId, teamInfo.domain, teamId)));
 
         const joinedUsers = ghosts.filter((g) => !existingGhosts.includes(g.userId)); // Skip users that are joined.
-        const leftUsers = existingGhosts.filter((userId) => !ghosts.find((g) => g.userId === userId ));
+        const leftUsers = existingGhosts.map((userId) => ghosts.find((g) => g.userId === userId )).filter(g => !!g) as SlackGhost[];
         log.info(`Joining ${joinedUsers.length} ghosts to ${roomId}`);
         log.info(`Leaving ${leftUsers.length} ghosts to ${roomId}`);
 
         const queue = new PQueue({concurrency: JOIN_CONCURRENCY});
 
         // Join users who aren't joined
-        joinedUsers.forEach(async(g) => queue.add(async() => g.intent.join(roomId)));
+        void queue.addAll(joinedUsers.map((ghost) => async () => {
+            try {
+                await ghost.intent.join(roomId);
+            } catch (ex) {
+                log.warn(`Failed to join ${ghost.userId} to ${roomId}`);
+            }
+        }));
 
         // Leave users who are joined
-        leftUsers.forEach(async(userId) => queue.add(async() => this.main.getIntent(userId).leave(roomId)));
+        void queue.addAll(leftUsers.map((ghost) => async () => {
```

Comment:
If we're sure that this will never fail and handling rejections is unnecessary, it should still be safer to `catch()` and fatal() what we get, in case the behaviour of `join()` or `leave()` changes in the future and causes this to silently eat errors.